src/HealthChecks.DynamoDb/DynamoDbHealthCheck.cs

```diff
@@ -23,7 +19,19 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         {
             try
             {
-                var credentials = new BasicAWSCredentials(_options.AccessKey, _options.SecretKey);
+                AWSCredentials credentials = _options.Credentials;
+                if (credentials == null)
+                {
+#pragma warning disable CS0618 // Type or member is obsolete
+                    if (!string.IsNullOrEmpty(_options.AccessKey) && !string.IsNullOrEmpty(_options.SecretKey))
+                    {
+                        // for backwards compatibility we create the basic credentials if the old fields are used
+                        // but if they are not specified we fallback to using the default profile

```

Comment:
i.e. `AmazonDynamoDBClient` will work properly with `null` value of `credentials`, right?

Reply:
This is the intention, yes. We wouldn't necessarily need to flag those as obsolete as they are supported just like before (but the Credentials field takes precedence). I figured people should be using BasicAwsCredentials to specify the access key and secret instead as it makes the intention more clear. But I could also see how that is a bit to big of a change to the API.

Reply:
Commit 9e7830f93a894ee9c5e4734b7f930133e31cce54 contains an example of how credentials would be specified using the BasicAWSCredentials (the point is that other kind of credentials could be specified instead of these though like AssumeAWSCredentials for instance)

Reply:
OK. Please also add `AddDynamoDb` overload that takes `Action<IServiceProvider, DynamoDBOptions>? setup` argument. Also you may find useful to add one more overload with `Func<>` delegate like `AddArangoDb`.

Reply:
Added a overload for service provider in commit 03cc53268644baf86b9f50429edbcd53262a9cac, I figured this was what you wanted?

Reply:
yes